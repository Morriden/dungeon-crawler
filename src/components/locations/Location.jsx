import React from 'react';
import PropTypes from 'prop-types';

const Location = ({ location }) => (
  <section>
    Title: {location.title}
    <img src={location.image} />
    Description: {location.description}
  </section>
);

Location.propTypes = {
  location: PropTypes.shape({
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
  })
};

export default Location;
