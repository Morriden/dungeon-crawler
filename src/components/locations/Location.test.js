import React from 'react';
import { shallow } from 'enzyme';
import Location from './Location';

describe('location component', () => {
  it('renders the location', () => {
    const location = {
      title: 'im a title',
      image: 'im an image',
      description: 'im an image'
    };
    const wrapper = shallow(<Location location={location} />);
    expect(wrapper).toMatchSnapshot();
  });
});
