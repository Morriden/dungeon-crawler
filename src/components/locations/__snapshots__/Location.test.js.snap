// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`location component renders the location 1`] = `
<section>
  Title: 
  im a title
  <img
    src="im an image"
  />
  Description: 
  im an image
</section>
`;
